amr-wind/wind_energy/actuator/disk/constant_ct_ops.H

```diff
@@ -0,0 +1,116 @@
+#ifndef CONSTANT_CT_OPS_H_
+#define CONSTANT_CT_OPS_H_
+
+#include "amr-wind/wind_energy/actuator/disk/ConstantCt.H"
+#include "amr-wind/wind_energy/actuator/actuator_ops.H"
+#include "amr-wind/utilities/tensor_ops.H"
+
+namespace amr_wind {
+namespace actuator {
+namespace ops {
+
+template <>
+struct ReadInputsOp<ConstantCt, ActSrcDisk>
+{
+    void operator()(ConstantCt::DataType& data, const utils::ActParser& pp)
+    {
+        auto& ddata = data.meta();
+        auto& info = data.info();
+        pp.get("num_points", ddata.num_points);
+        pp.get("disk_center", ddata.center);
+        pp.get("disk_normal", ddata.normal);
+        pp.get("epsilon", ddata.epsilon);
+        pp.get("diameter", ddata.diameter);
+        // TODO think about center of disk
+        const amrex::Real radius = ddata.diameter * 0.5;
+        ddata.dr = radius / ddata.num_points;
+
+        // ensure normal is normalized
+        ddata.normal = ddata.normal.normalize();
+        const vs::Vector& n = ddata.normal;
+        // get a vector in the plane of the disk
+        vs::Vector planeVec = vs::Vector::one();
+        const auto unit = vs::Tensor::I();
+
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+            std::abs((unit.z() & n) - vs::mag_sqr(n)) >
+                vs::DTraits<amrex::Real>::eps(),
+            "A constant Ct disk normal is too close to vertical. Vertical axis "
+            "wind turbines are not supported with this model.");
+
+        ::amr_wind::utils::cross_prod(
+            n.data(), unit.z().data(), planeVec.data());
+
+        // clang-format off
+        const amrex::Real nl = ddata.epsilon * 3.0; // length scale in normal dir
+        const amrex::Real dl = radius + ddata.dr*2.0; // length scale in plane of disk
+
+        const auto& cc = ddata.center;
+        const amrex::Real xLen = (unit.x()&n)*nl + (unit.x()&planeVec)*dl;
+        const amrex::Real yLen = (unit.y()&n)*nl + (unit.y()&planeVec)*dl;
+        const amrex::Real zLen = (unit.z()&n)*nl + (unit.z()&planeVec)*dl;
+
+        info.bound_box = amrex::RealBox(
+          cc.x()-xLen, cc.y()-yLen, cc.z()-zLen, 
+          cc.x()+xLen, cc.y()+yLen, cc.z()+zLen
+        );
```

Comment:
This doesn't look correct to me. Let us say

```
n = (1 0 0)
unit.z() = (0 0 1)
planeVec = n ^ z
``` 

Then zLen works out to be zero right?

Reply:
Try this 

```C++
const auto dvec = n * nl + planeVec * dl + vs::Vector::khat() * dl;
const auto p1 = cc - dvec;  // front
const auto p2 = cc + dvec;  // back
info.bound_box = amrex::RealBox(
    amrex::min(p1.x(), p2.x()),
    amrex::min(p1.y(), p2.y()),
    amrex::min(p1.z(), p2.z()),
    amrex::max(p1.x(), p2.x()),
    amrex::max(p1.y(), p2.y()),
    amrex::max(p1.z(), p2.z())
);
```